tokio/src/sync/semaphore.rs

```diff
@@ -123,6 +123,79 @@ use std::sync::Arc;
 /// # }
 /// ```
 ///
+/// ## Prevent tests from running in parallel
+///
+/// By default, Rust parallelly runs tests in the same file. However, In some cases, running two tests in parallel may lead to some problems.
+/// For example, this can happen when tests use the same database.
+///
+/// Consider the following scenario:
+/// 1. `test_insert`: Inserts a key-value pair into the database, then retrieves the value using the same key to verify the insertion.
+/// 2. `test_update`: After insertion, this test updates the key's value to a new one and verifies if the value has been accurately updated.
+/// 3. `test_others`: Other tests that dosen't modify the database state, thus can run parallely with previous tests.
+///
+/// For this example, `test_insert` and `test_update` need to run in sequence to ensure they can consistently pass., but it's doesn't matter which one runs first.
+/// Leveraging a semaphore with a single permit elegantly addresses this challenge.
```

Comment:
Modifying this to use active voice makes it easier to read.
```suggestion
/// In this example, `test_insert` and `test_update` need to run in sequence to work, but it doesn't matter which test runs first.
/// We can leverage a semaphore with a single permit to address this challenge.
```

Reply:
Here I want to emphasis that insert and update test may work well in parallel for most time, but in some case, say test_insert gets the updated value,it won't work. So I want to say something like: you need semaphore to make sure that insert and update tests will always pass.